doc/ss_logger.rst

```diff
@@ -0,0 +1,144 @@
+associated logging functions users may use within an experiment driver script:
+
+    1. level: `quiet`
+       - function: ``logger.error()`` & ``logger.warning()``
+    2. level: `info`
+       - function: ``logger.info()``
+    3. level: `debug`
+       - function: ``logger.debug()``
+
+The messages printed to stdout or to file by the SmartSim logger depend on the specified value.
+The logger prints the specified severity level and all levels below it. For instance,
+if `SMARTSIM_LOG_LEVEL=info`, the logger will print messages from ``logger.error()``,
+``logger.warning()``, and ``logger.info()``. With `SMARTSIM_LOG_LEVEL=debug`, messages from
+``logger.error()``, ``logger.warning()``, ``logger.info()``, and ``logger.debug()`` will be printed.
+
+==========================
+Example: Logging to stdout
+==========================
+Below, we provide an implementation of the SmartSim logger.
+
+To use the SmartSim logger, import the required module:
+
+.. code-block:: python
+
+      from smartsim.log import get_logger
+
+Next, initialize an instance of the logger and provide a `name`:
+
+.. code-block:: python
+
+      logger = get_logger("example_logger")
+
+To demonstrate the full functionality of of the SmartSim logger, we include all log
+functions in the Python driver script with log messages:
+
+.. code-block:: python
+
+      logger.info("This is a message")
+      logger.debug("This is a debug message")
+      logger.error("This is an error message")
+      logger.warning("This is a warning message")
+
+Before executing the Python script, we must set the environment variable, `SMARTSIM_LOG_LEVEL`,
+to a log level. The default is `info`.
+For the example, set `SMARTSIM_LOG_LEVEL` to `quiet` in the terminal::
+    export SMARTSIM_LOG_LEVEL=quiet
+
+When we execute the script with `SMARTSIM_LOG_LEVEL=quiet`,
+the following messages from will print from above::
+    21:07:40 osprey.us.cray.com SmartSim[10950] ERROR This is an error message
+    21:07:40 osprey.us.cray.com SmartSim[10950] WARNING This is a warning message
+
+Notice that the `info` and `debug` messages were ignored. This is because by setting
+a lower severity level (`quiet`), we instruct SmartSim to ignore the higher severity levels (`debug` and `info`).
+
+Next, set `SMARTSIM_LOG_LEVEL` to `info` in the terminal::
+    export SMARTSIM_LOG_LEVEL=info
+
+When we run the script with `SMARTSIM_LOG_LEVEL=info`,
+the following output appears in stdout::
+    19:52:05 osprey.us.cray.com SmartSim[130033] INFO This is a message
+    19:52:05 osprey.us.cray.com SmartSim[130033] ERROR This is an error message
+    19:52:05 osprey.us.cray.com SmartSim[130033] WARNING This is a warning message
+
+Above, we have instructed SmartSim to print all `info` messages but ignore higher
+verbose messages (`debug`).
+
+Now set the `SMARTSIM_LOG_LEVEL` to `debug` and check the output of the program.
+Set the environment variable in the terminal like so::
```

Comment:
I'd get rid of "like so" and review instructions in each example to make sure all instructions follow same format. 

Consider modifying the info example to show that an unset env var results in the same output